components/script/dom/serviceworkercontainer.rs

```diff
@@ -57,6 +66,26 @@ impl ServiceWorkerContainerMethods for ServiceWorkerContainer {
     fn Register(&self,
                 script_url: USVString,
                 options: &RegistrationOptions) -> Fallible<Root<ServiceWorkerRegistration>> {
+        
+        
+        /*Let p be a promise.
+        Let client be the context object’s service worker client.
+        Let scriptURL be the result of parsing scriptURL with the context object’s relevant settings object’s API base URL.
+        Let scopeURL be null.
+        If options.scope is present, set scopeURL to the result of parsing options.scope with the context object’s relevant settings object’s API base URL.
+        Invoke Start Register with scopeURL, scriptURL, p, client, client’s creation URL and options.type.
+        Return p.    */ 
+        
+        // New Step 1
+        
+        let p = dom::promise::Promise
```

Comment:
I recommend looking for uses of `Promise::new` in the codebase to base your changes off of.